modules/aws/eks/eks.tf

```diff
@@ -51,6 +54,63 @@ resource "aws_eks_cluster" "this" {
     }
   }
 
+  tags       = local.global_tags
+  depends_on = [aws_cloudwatch_log_group.this]
+}
+
+
+
+# CloudWatch Encrytion
+resource "aws_kms_key" "cloudwatch_encryption" {
+  description         = "Used for cloudwatch encryption"
+  enable_key_rotation = true
+
+  policy = <<EOF
+{
+  "Version" : "2012-10-17",
+  "Id" : "key-default-1",
+  "Statement" : [ {
+      "Sid" : "Enable IAM User Permissions",
+      "Effect" : "Allow",
+      "Principal" : {
+        "AWS" : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      },
+      "Action" : "kms:*",
+      "Resource" : "*"
+    },
+    {
+      "Effect": "Allow",
+      "Principal": { "Service": "logs.${data.aws_region.current.name}.amazonaws.com" },
+      "Action": [
+        "kms:Encrypt*",
+        "kms:Decrypt*",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:Describe*"
```

Comment:
I imagine you copied these from AWS docs? AWS docs are infamous for crappy IAM practices. Among other things: there is only `kms:Encrypt` and `kms:Decrypt` so those stars are spurious. I suspect you can reduce `kms:Describe*` to `kms:DescribeKey`. (ReEncrypt* and GenerateDataKey* are legit.) Details here https://docs.aws.amazon.com/service-authorization/latest/reference/list_awskeymanagementservice.html#awskeymanagementservice-actions-as-permissions .